test/mongooseim_metrics_SUITE.erl

```diff
@@ -108,9 +113,24 @@ up_time_positive(_C) ->
     {ok, [{value, X}]} = mongoose_metrics:get_metric_value(global, nodeUpTime),
     ?assert(X > 0).
 
+get_new_tcp_metric_value(OldValue) ->
+    Validator = fun(NewValue) -> OldValue =/= NewValue end,
+    {ok, {ok, [{value, X}]}} = async_helper:wait_until(
+      fun() -> mongoose_metrics:get_metric_value(global, tcpPortsUsed) end,
+      true, #{validator => Validator, sleep_time => 30, time_left => 500}
+     ),
+    X.
+
 tcp_connections_detected(_C) ->
-    {ok, [{value, X}]} = mongoose_metrics:get_metric_value(global, tcpPortsUsed),
-    ?assert(X > 0).
+    get_new_tcp_metric_value({ok, []}).
+
+tcp_metric_varies_with_tcp_variations(_C) ->
+    X = get_new_tcp_metric_value({ok, []}),
+    {ok, Socket} = gen_tcp:listen(1805, []),
```

Comment:
I'm not sure the metric works as documented. In the doc I can see:

>  A number of open tcp connections. This should relate to the number of connected sessions and databases, in order to detect connection leaks. 

And in this test we start listening on a port which also increments the counter. How does it correlated to the what's in the doc?

Reply:
That both sessions and databases connect through tcp ports, for example the sessions through `ejabberd_c2s` and `ejabberd_receiver`, and the databases through worker pools (maybe @erszcz can say something more about it?).

Reply:
Yes, and there is more:
* s2s incoming connections
* s2s outgoing connections
* connections for HTTP request MongooseIM sends
* connections for HTTP request MongooseIM serves

Basically any incoming or outgoing TCP connection as well as TCP ports MongooseIM listens on (without any connection at a particular moment). If this is what we want, that's fine and I think the doc could be improved as currently it may be misleading.